src/main/java/ch/njol/skript/expressions/ExprDyed.java

```diff
@@ -0,0 +1,128 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.Color;
+import ch.njol.util.Kleenean;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.inventory.meta.LeatherArmorMeta;
+import org.bukkit.inventory.meta.MapMeta;
+import org.bukkit.inventory.meta.PotionMeta;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.regex.Matcher;
+
+@Name("Dyed")
+@Description("An expression to return items/entities with a color.")
+@Examples({
+		"give player leather chestplate dyed red",
+		"give player potion of invisibility dyed rgb 200, 70, 88",
+		"give player filled map with color rgb(20, 60, 70)"
+})
+@Since("INSERT VERSION")
+public class ExprDyed extends SimpleExpression<ItemType> {
+
+	private static final boolean MAPS_AND_POTIONS_COLORS = Skript.methodExists(PotionMeta.class, "setColor", org.bukkit.Color.class);
+	
+	static {
+		Skript.registerExpression(ExprDyed.class, ItemType.class, ExpressionType.COMBINED, "%itemtypes% (dyed|with color [of]) %color%");
```

Comment:
I don't think the with color of fits this syntax.
```suggestion
		Skript.registerExpression(ExprDyed.class, ItemType.class, ExpressionType.COMBINED, "%itemtypes% (dyed|colo[u]red) %color%");
```

Reply:
I can add `colo[u]red` but I prefer not to remove `with color of` as I was following **ExprNamed**
https://github.com/SkriptLang/Skript/blob/8c7fec07d75990eb67b7bf244c81babea4ba2585/src/main/java/ch/njol/skript/expressions/ExprNamed.java#L56

Reply:
I don't know how I feel about `with color` being valid syntax as `leather chestplate with coloured blue` doesn't make sense 
`leather chestplate coloured blue` makes a bit more sense but overall I prefer just `dyed` `leather chestplate dyed green`.

Only other time I see `coloured` being good is `leather helmet coloured rgb(10,10,10)`

In this case I agree with Lime's opinion on syntax

Reply:
The syntax doesn't allow `with colored [of]` tho (`ed`) it's only `with color [of]` or `dyed` or `colo[u]red`

Reply:
Yeah I fixed my spelling issue there, but none the less `with color` doesn't make sense to me

Reply:
How's that if I may know? `leather boots with color red`, `leather boots dyed red` and `leather boots colored red`
We also have `leather boots named "test"` and `leather boots with name "test"`

Reply:
1. the `with name` is more often than not never used against itemtypes but that still makes sense in cases.
2. `with color` just doesn't in my eyes explain hey this is item will be coloured green red, blue whatever they want. Where as `colo[u]red %color%` states this item right here is coloured this. 

The usage of `with` implies to me that this will be included with it and not as apart of it which color is apart of the item.
*sorry can't think of a better way to explain it*

While skript does actively use `with` for anything related to item meta, we don't do that for enchantments, probably because we still never bothered to remove it from it's literal limitation.

I'd like to mention when ever you explain what color something is you don't say "Hey your shirt is with color teal" but "Hey your shirt is colored teal, I dig that".  I just don't see `with color` being valid grammar wise. I am by no means good with proper grammar as known by all my rants in discord.

I believe this is something we should have more conversation about from others preferably Sovde as I know he's good with grammar type things.

---

Little side option opinion on `%*color% %itemtype%`? `red leather chestplate`, `green leather boots`? Crossed my mind that some sort of literal option could work here.

Reply:
I get the argument about parity with named and it's not a bad one, I generally agree. I would be ok with leaving in `with colour` for that reason.

That said, I don't really like it. From a grammar/usage perspective, `with name "xyz"` isn't great but `with the name "xyz"` works perfectly fine to describe something having a name `"It's common to have a pet with the name Fido."` That doesn't hold true for colour, in my experience. I can't think of a single instance where `"a subject with the colour blue"` feels natural to say. The only examples I can think of are when you're talking about the colour itself as the subject, not as a property of another subject: `"i have issues with the colour blue"`.

 The closest I can get is `"choose the blocks with the colour blue"` which does actually use blue as the property of the blocks, but it also has a weird aspect of being like a filter, not a property. I feel like I'm lacking the proper English Major jargon to describe this properly, but suffice it to say that I would never use `"with the colour blue"` as a standalone way to ascribe a property to an object, only as a filter to select some of a group. I'd only use things like `"a shovel with the colour blue on it"`, but even then I'd much rather use `"a shovel painted blue"` or `"a shovel [that's] coloured blue"`. Perhaps I'm drawing an unfair comparison and `"It's common to have a pet with the name Fido."` should also be treated this way, but at this point I'm running on vibes.

Having said all this, I'd like to sum up by saying that I don't like `with colour ...`, but it's not something I'd actively block from Skript. Plenty of syntaxes are on the wrong side of grammatical or natural but feel fine in Skript, and I think the parity with `with name` is a good enough argument. **I think `with colour` is a worse syntax than `dyed` or `coloured`, but it's not bad enough to block.**

Reply:
> Little side option opinion on %*color% %itemtype%? red leather chestplate, green leather boots? Crossed my mind that some sort of literal option could work here.

That's very much gonna cause either parser conflicts or user conflict (as aliases) so I would strongly vote no on that.


As for the opinions on `with color`, thank you all for the detailed explanation, I appreciate that and thus, please let me know your vote on the following:
1- Keep `with [the] color` just because we `with [the] name` (`[the]` part will be added to both)
2- Remove `with color` and add `painted` instead

Thanks in advance.

Reply:
> That said, I don't really like it. From a grammar/usage perspective, `with name "xyz"` isn't great but `with the name "xyz"` works perfectly fine to describe something having a name `"It's common to have a pet with the name Fido."` That doesn't hold true for colour, in my experience. I can't think of a single instance where `"a subject with the colour blue"` feels natural to say.

The correct (although archaic) version is *in* the colour blue.

Reply:
I don't like `with [the] color` :(